### PR TITLE
followup the PR#32701 and fixed issues based on comments left by @hzx…

### PR DESCRIPTION
**Pull request by zhlsunshine**
This PR is submitted for following up previous [PR32701](https://github.com/istio/istio/pull/32701), I fixed issues based on comments left by @hzxuzhonghu, moreover, there is new enhancement on overlapping match validation for virtual service so I added 2 unit tests for `TestValidateVirtualService` which was required by @hzxuzhonghu 

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Comment by zhlsunshine** [comment]
Hi @hzxuzhonghu, I have checked the `make test` and functionality for this new PR, the result is below:

**Unit tests validation:**

```
$ make test
make test
useradd: UID 0 is not unique
go test   -race ./... 2>&1 | tee >(/usr/bin/go-junit-report > /work/out/linux_amd64/junit.xml)
?   	istio.io/istio/cni/cmd/install-cni	[no test files]
ok  	istio.io/istio/cni/cmd/istio-cni	(cached)
?   	istio.io/istio/cni/cmd/istio-cni-repair	[no test files]
?   	istio.io/istio/cni/cmd/istio-cni-taint	[no test files]
?   	istio.io/istio/cni/pkg/install-cni/cmd	[no test files]
?   	istio.io/istio/cni/pkg/install-cni/pkg/config	[no test files]
?   	istio.io/istio/cni/pkg/install-cni/pkg/constants	[no test files]
ok  	istio.io/istio/cni/pkg/install-cni/pkg/install	(cached)
?   	istio.io/istio/cni/pkg/install-cni/pkg/util	[no test files]
ok  	istio.io/istio/cni/pkg/repair	(cached)
ok  	istio.io/istio/cni/pkg/taint	(cached)
ok  	istio.io/istio/cni/test	(cached)
ok  	istio.io/istio/galley/pkg/config/analysis	(cached)
ok  	istio.io/istio/galley/pkg/config/analysis/analyzers	(cached)
?   	istio.io/istio/galley/pkg/config/analysis/analyzers/annotations	[no test files]
ok  	istio.io/istio/galley/pkg/config/analysis/analyzers/authz	(cached)
?   	istio.io/istio/galley/pkg/config/analysis/analyzers/deployment	[no test files]
?   	istio.io/istio/galley/pkg/config/analysis/analyzers/deprecation	[no test files]
?   	istio.io/istio/galley/pkg/config/analysis/analyzers/destinationrule	[no test files]
?   	istio.io/istio/galley/pkg/config/analysis/analyzers/gateway	[no test files]
?   	istio.io/istio/galley/pkg/config/analysis/analyzers/injection	[no test files]
?   	istio.io/istio/galley/pkg/config/analysis/analyzers/multicluster	[no test files]
ok  	istio.io/istio/galley/pkg/config/analysis/analyzers/schema	(cached)
?   	istio.io/istio/galley/pkg/config/analysis/analyzers/service	[no test files]
?   	istio.io/istio/galley/pkg/config/analysis/analyzers/serviceentry	[no test files]
?   	istio.io/istio/galley/pkg/config/analysis/analyzers/sidecar	[no test files]
ok  	istio.io/istio/galley/pkg/config/analysis/analyzers/util	(cached)
?   	istio.io/istio/galley/pkg/config/analysis/analyzers/virtualservice	[no test files]
?   	istio.io/istio/galley/pkg/config/analysis/analyzers/webhook	[no test files]
ok  	istio.io/istio/galley/pkg/config/analysis/diag	(cached)
ok  	istio.io/istio/galley/pkg/config/analysis/local	(cached)
?   	istio.io/istio/galley/pkg/config/analysis/msg	[no test files]
?   	istio.io/istio/galley/pkg/config/analysis/testing/fixtures	[no test files]
ok  	istio.io/istio/galley/pkg/config/collection	(cached)
ok  	istio.io/istio/galley/pkg/config/mesh	(cached)
?   	istio.io/istio/galley/pkg/config/monitoring	[no test files]
ok  	istio.io/istio/galley/pkg/config/processing	(cached)
ok  	istio.io/istio/galley/pkg/config/processing/snapshotter	(cached)
ok  	istio.io/istio/galley/pkg/config/processing/snapshotter/strategy	(cached)
ok  	istio.io/istio/galley/pkg/config/processing/transformer	(cached)
ok  	istio.io/istio/galley/pkg/config/processor	(cached)
ok  	istio.io/istio/galley/pkg/config/processor/groups	(cached)
?   	istio.io/istio/galley/pkg/config/processor/transforms	[no test files]
ok  	istio.io/istio/galley/pkg/config/processor/transforms/direct	(cached)
?   	istio.io/istio/galley/pkg/config/scope	[no test files]
ok  	istio.io/istio/galley/pkg/config/source/inmemory	(cached)
ok  	istio.io/istio/galley/pkg/config/source/kube	(cached)
ok  	istio.io/istio/galley/pkg/config/source/kube/apiserver	(cached)
?   	istio.io/istio/galley/pkg/config/source/kube/apiserver/stats	[no test files]
ok  	istio.io/istio/galley/pkg/config/source/kube/apiserver/status	(cached)
ok  	istio.io/istio/galley/pkg/config/source/kube/apiserver/tombstone	(cached)
ok  	istio.io/istio/galley/pkg/config/source/kube/inmemory	(cached)
ok  	istio.io/istio/galley/pkg/config/source/kube/rt	(cached)
ok  	istio.io/istio/galley/pkg/config/source/mcp	(cached)
?   	istio.io/istio/galley/pkg/config/testing/basicmeta	[no test files]
?   	istio.io/istio/galley/pkg/config/testing/data	[no test files]
ok  	istio.io/istio/galley/pkg/config/testing/fixtures	(cached)
?   	istio.io/istio/galley/pkg/config/testing/k8smeta	[no test files]
?   	istio.io/istio/galley/pkg/config/util/kuberesource	[no test files]
ok  	istio.io/istio/galley/pkg/config/util/kubeyaml	(cached)
ok  	istio.io/istio/galley/pkg/config/util/pb	(cached)
ok  	istio.io/istio/galley/pkg/server/components	(cached)
ok  	istio.io/istio/galley/pkg/server/settings	(cached)
?   	istio.io/istio/galley/pkg/testing/mock	[no test files]
?   	istio.io/istio/galley/testdatasets/validation	[no test files]
ok  	istio.io/istio/istioctl/cmd	10.817s
ok  	istio.io/istio/istioctl/cmd/istioctl	0.653s
?   	istio.io/istio/istioctl/pkg/authz	[no test files]
?   	istio.io/istio/istioctl/pkg/clioptions	[no test files]
?   	istio.io/istio/istioctl/pkg/install	[no test files]
ok  	istio.io/istio/istioctl/pkg/install/k8sversion	(cached)
ok  	istio.io/istio/istioctl/pkg/multicluster	1.384s
?   	istio.io/istio/istioctl/pkg/multixds	[no test files]
?   	istio.io/istio/istioctl/pkg/util/clusters	[no test files]
?   	istio.io/istio/istioctl/pkg/util/configdump	[no test files]
ok  	istio.io/istio/istioctl/pkg/util/formatting	(cached)
ok  	istio.io/istio/istioctl/pkg/util/handlers	(cached)
?   	istio.io/istio/istioctl/pkg/util/proto	[no test files]
ok  	istio.io/istio/istioctl/pkg/validate	(cached)
ok  	istio.io/istio/istioctl/pkg/verifier	(cached)
ok  	istio.io/istio/istioctl/pkg/writer/compare	(cached) [no tests to run]
ok  	istio.io/istio/istioctl/pkg/writer/compare/sds	(cached)
?   	istio.io/istio/istioctl/pkg/writer/envoy/clusters	[no test files]
ok  	istio.io/istio/istioctl/pkg/writer/envoy/configdump	(cached)
ok  	istio.io/istio/istioctl/pkg/writer/pilot	(cached)
?   	istio.io/istio/istioctl/pkg/xds	[no test files]
?   	istio.io/istio/manifests	[no test files]
?   	istio.io/istio/operator/cmd	[no test files]
ok  	istio.io/istio/operator/cmd/mesh	(cached)
?   	istio.io/istio/operator/cmd/operator	[no test files]
?   	istio.io/istio/operator/pkg/apis	[no test files]
?   	istio.io/istio/operator/pkg/apis/istio	[no test files]
?   	istio.io/istio/operator/pkg/apis/istio/fixup_structs	[no test files]
ok  	istio.io/istio/operator/pkg/apis/istio/v1alpha1	(cached)
ok  	istio.io/istio/operator/pkg/apis/istio/v1alpha1/validation	(cached)
ok  	istio.io/istio/operator/pkg/cache	(cached)
ok  	istio.io/istio/operator/pkg/compare	(cached)
?   	istio.io/istio/operator/pkg/component	[no test files]
?   	istio.io/istio/operator/pkg/controller	[no test files]
?   	istio.io/istio/operator/pkg/controller/istiocontrolplane	[no test files]
ok  	istio.io/istio/operator/pkg/controlplane	(cached)
ok  	istio.io/istio/operator/pkg/helm	(cached)
ok  	istio.io/istio/operator/pkg/helmreconciler	(cached)
ok  	istio.io/istio/operator/pkg/httprequest	(cached)
ok  	istio.io/istio/operator/pkg/manifest	(cached)
?   	istio.io/istio/operator/pkg/metrics	[no test files]
ok  	istio.io/istio/operator/pkg/name	(cached)
ok  	istio.io/istio/operator/pkg/object	(cached)
ok  	istio.io/istio/operator/pkg/patch	(cached)
ok  	istio.io/istio/operator/pkg/tpath	(cached)
ok  	istio.io/istio/operator/pkg/translate	(cached)
ok  	istio.io/istio/operator/pkg/util	(cached)
?   	istio.io/istio/operator/pkg/util/clog	[no test files]
?   	istio.io/istio/operator/pkg/util/httpserver	[no test files]
ok  	istio.io/istio/operator/pkg/util/progress	(cached)
?   	istio.io/istio/operator/pkg/util/tgz	[no test files]
ok  	istio.io/istio/operator/pkg/validate	(cached)
ok  	istio.io/istio/operator/pkg/version	(cached)
?   	istio.io/istio/operator/version	[no test files]
ok  	istio.io/istio/pilot/cmd/pilot-agent	(cached)
ok  	istio.io/istio/pilot/cmd/pilot-agent/config	(cached)
?   	istio.io/istio/pilot/cmd/pilot-agent/metrics	[no test files]
ok  	istio.io/istio/pilot/cmd/pilot-agent/options	(cached)
ok  	istio.io/istio/pilot/cmd/pilot-agent/status	(cached)
ok  	istio.io/istio/pilot/cmd/pilot-agent/status/ready	(cached)
?   	istio.io/istio/pilot/cmd/pilot-agent/status/testserver	[no test files]
?   	istio.io/istio/pilot/cmd/pilot-agent/status/util	[no test files]
?   	istio.io/istio/pilot/cmd/pilot-discovery	[no test files]
?   	istio.io/istio/pilot/cmd/pilot-discovery/app	[no test files]
ok  	istio.io/istio/pilot/pkg/bootstrap	(cached)
ok  	istio.io/istio/pilot/pkg/config/aggregate	(cached)
ok  	istio.io/istio/pilot/pkg/config/kube/crd	(cached)
ok  	istio.io/istio/pilot/pkg/config/kube/crdclient	(cached)
?   	istio.io/istio/pilot/pkg/config/kube/crdclient/gen	[no test files]
ok  	istio.io/istio/pilot/pkg/config/kube/gateway	(cached)
ok  	istio.io/istio/pilot/pkg/config/kube/ingress	(cached)
ok  	istio.io/istio/pilot/pkg/config/kube/ingressv1	(cached)
ok  	istio.io/istio/pilot/pkg/config/memory	(cached)
ok  	istio.io/istio/pilot/pkg/config/monitor	(cached)
ok  	istio.io/istio/pilot/pkg/controller/workloadentry	(cached)
ok  	istio.io/istio/pilot/pkg/dns	(cached)
?   	istio.io/istio/pilot/pkg/extensionproviders	[no test files]
?   	istio.io/istio/pilot/pkg/features	[no test files]
ok  	istio.io/istio/pilot/pkg/grpc	(cached)
ok  	istio.io/istio/pilot/pkg/keycertbundle	(cached)
ok  	istio.io/istio/pilot/pkg/leaderelection	(cached)
ok  	istio.io/istio/pilot/pkg/model	(cached)
?   	istio.io/istio/pilot/pkg/model/kstatus	[no test files]
?   	istio.io/istio/pilot/pkg/model/status	[no test files]
?   	istio.io/istio/pilot/pkg/model/test	[no test files]
ok  	istio.io/istio/pilot/pkg/networking	(cached)
ok  	istio.io/istio/pilot/pkg/networking/apigen	(cached)
?   	istio.io/istio/pilot/pkg/networking/core	[no test files]
ok  	istio.io/istio/pilot/pkg/networking/core/v1alpha3	(cached)
ok  	istio.io/istio/pilot/pkg/networking/core/v1alpha3/envoyfilter	(cached)
ok  	istio.io/istio/pilot/pkg/networking/core/v1alpha3/loadbalancer	(cached)
ok  	istio.io/istio/pilot/pkg/networking/core/v1alpha3/route	(cached)
ok  	istio.io/istio/pilot/pkg/networking/core/v1alpha3/route/retry	(cached)
ok  	istio.io/istio/pilot/pkg/networking/grpcgen	(cached)
?   	istio.io/istio/pilot/pkg/networking/plugin	[no test files]
ok  	istio.io/istio/pilot/pkg/networking/plugin/authn	(cached)
?   	istio.io/istio/pilot/pkg/networking/plugin/authz	[no test files]
?   	istio.io/istio/pilot/pkg/networking/plugin/metadataexchange	[no test files]
ok  	istio.io/istio/pilot/pkg/networking/plugin/registry	(cached)
ok  	istio.io/istio/pilot/pkg/networking/util	(cached)
?   	istio.io/istio/pilot/pkg/proto	[no test files]
ok  	istio.io/istio/pilot/pkg/request	(cached)
?   	istio.io/istio/pilot/pkg/secrets	[no test files]
ok  	istio.io/istio/pilot/pkg/secrets/kube	(cached)
?   	istio.io/istio/pilot/pkg/security/authn	[no test files]
?   	istio.io/istio/pilot/pkg/security/authn/factory	[no test files]
?   	istio.io/istio/pilot/pkg/security/authn/utils	[no test files]
ok  	istio.io/istio/pilot/pkg/security/authn/v1beta1	(cached)
ok  	istio.io/istio/pilot/pkg/security/authz/builder	(cached)
ok  	istio.io/istio/pilot/pkg/security/authz/matcher	(cached)
ok  	istio.io/istio/pilot/pkg/security/authz/model	(cached)
ok  	istio.io/istio/pilot/pkg/security/model	(cached)
ok  	istio.io/istio/pilot/pkg/security/trustdomain	(cached)
ok  	istio.io/istio/pilot/pkg/server	(cached)
ok  	istio.io/istio/pilot/pkg/serviceregistry	(cached)
ok  	istio.io/istio/pilot/pkg/serviceregistry/aggregate	(cached)
ok  	istio.io/istio/pilot/pkg/serviceregistry/kube	(cached)
ok  	istio.io/istio/pilot/pkg/serviceregistry/kube/controller	(cached)
?   	istio.io/istio/pilot/pkg/serviceregistry/kube/controller/filter	[no test files]
?   	istio.io/istio/pilot/pkg/serviceregistry/memory	[no test files]
ok  	istio.io/istio/pilot/pkg/serviceregistry/mock	(cached)
ok  	istio.io/istio/pilot/pkg/serviceregistry/serviceentry	(cached)
?   	istio.io/istio/pilot/pkg/simulation	[no test files]
ok  	istio.io/istio/pilot/pkg/status	(cached)
ok  	istio.io/istio/pilot/pkg/trustbundle	(cached)
?   	istio.io/istio/pilot/pkg/util/informermetric	[no test files]
ok  	istio.io/istio/pilot/pkg/util/network	(cached)
ok  	istio.io/istio/pilot/pkg/util/runtime	(cached)
ok  	istio.io/istio/pilot/pkg/util/sets	(cached)
ok  	istio.io/istio/pilot/pkg/xds	(cached)
?   	istio.io/istio/pilot/pkg/xds/filters	[no test files]
?   	istio.io/istio/pilot/pkg/xds/v3	[no test files]
?   	istio.io/istio/pilot/test/grpcgen	[no test files]
?   	istio.io/istio/pilot/test/mock	[no test files]
?   	istio.io/istio/pilot/test/util	[no test files]
?   	istio.io/istio/pilot/test/xdstest	[no test files]
?   	istio.io/istio/pilot/tools/debug	[no test files]
ok  	istio.io/istio/pkg/adsc	(cached)
ok  	istio.io/istio/pkg/bootstrap	(cached)
ok  	istio.io/istio/pkg/bootstrap/option	(cached)
ok  	istio.io/istio/pkg/bootstrap/platform	(cached)
ok  	istio.io/istio/pkg/cmd	(cached)
ok  	istio.io/istio/pkg/config	(cached)
?   	istio.io/istio/pkg/config/constants	[no test files]
ok  	istio.io/istio/pkg/config/crd	(cached)
ok  	istio.io/istio/pkg/config/event	(cached)
?   	istio.io/istio/pkg/config/gateway	[no test files]
ok  	istio.io/istio/pkg/config/host	(cached)
?   	istio.io/istio/pkg/config/kube	[no test files]
ok  	istio.io/istio/pkg/config/labels	(cached)
ok  	istio.io/istio/pkg/config/mesh	(cached)
ok  	istio.io/istio/pkg/config/mesh/kubemesh	(cached)
ok  	istio.io/istio/pkg/config/protocol	(cached)
ok  	istio.io/istio/pkg/config/resource	(cached)
ok  	istio.io/istio/pkg/config/schema	(cached)
ok  	istio.io/istio/pkg/config/schema/ast	(cached)
ok  	istio.io/istio/pkg/config/schema/codegen	(cached)
ok  	istio.io/istio/pkg/config/schema/collection	(cached)
ok  	istio.io/istio/pkg/config/schema/collections	(cached)
?   	istio.io/istio/pkg/config/schema/fuzz	[no test files]
?   	istio.io/istio/pkg/config/schema/gvk	[no test files]
ok  	istio.io/istio/pkg/config/schema/resource	(cached)
?   	istio.io/istio/pkg/config/schema/snapshots	[no test files]
ok  	istio.io/istio/pkg/config/security	(cached)
ok  	istio.io/istio/pkg/config/validation	(cached)
?   	istio.io/istio/pkg/config/visibility	[no test files]
?   	istio.io/istio/pkg/config/xds	[no test files]
ok  	istio.io/istio/pkg/envoy	5.857s
?   	istio.io/istio/pkg/envoy/config/authentication/v1alpha1	[no test files]
?   	istio.io/istio/pkg/envoy/config/filter/http/alpn/v2alpha1	[no test files]
?   	istio.io/istio/pkg/envoy/config/filter/http/authn/v2alpha1	[no test files]
?   	istio.io/istio/pkg/envoy/config/filter/network/tcp_cluster_rewrite/v2alpha1	[no test files]
?   	istio.io/istio/pkg/errdict	[no test files]
?   	istio.io/istio/pkg/file	[no test files]
ok  	istio.io/istio/pkg/istio-agent	(cached)
ok  	istio.io/istio/pkg/istio-agent/health	(cached)
?   	istio.io/istio/pkg/istio-agent/metrics	[no test files]
?   	istio.io/istio/pkg/jwt	[no test files]
ok  	istio.io/istio/pkg/keepalive	(cached)
ok  	istio.io/istio/pkg/kube	1.431s
?   	istio.io/istio/pkg/kube/apimirror	[no test files]
ok  	istio.io/istio/pkg/kube/configmapwatcher	(cached)
ok  	istio.io/istio/pkg/kube/inject	(cached)
ok  	istio.io/istio/pkg/kube/secretcontroller	(cached)
ok  	istio.io/istio/pkg/listwatch	(cached)
ok  	istio.io/istio/pkg/mcp/internal	(cached)
?   	istio.io/istio/pkg/mcp/internal/test	[no test files]
?   	istio.io/istio/pkg/mcp/sink	[no test files]
ok  	istio.io/istio/pkg/mcp/snapshot	(cached)
?   	istio.io/istio/pkg/mcp/source	[no test files]
?   	istio.io/istio/pkg/mcp/status	[no test files]
?   	istio.io/istio/pkg/mcp/testing/groups	[no test files]
?   	istio.io/istio/pkg/proto	[no test files]
?   	istio.io/istio/pkg/proxy	[no test files]
ok  	istio.io/istio/pkg/queue	(cached)
?   	istio.io/istio/pkg/security	[no test files]
ok  	istio.io/istio/pkg/spiffe	(cached)
ok  	istio.io/istio/pkg/test	(cached)
?   	istio.io/istio/pkg/test/cert	[no test files]
?   	istio.io/istio/pkg/test/cert/ca	[no test files]
?   	istio.io/istio/pkg/test/config	[no test files]
?   	istio.io/istio/pkg/test/echo/client	[no test files]
?   	istio.io/istio/pkg/test/echo/cmd/client	[no test files]
?   	istio.io/istio/pkg/test/echo/cmd/server	[no test files]
?   	istio.io/istio/pkg/test/echo/common	[no test files]
?   	istio.io/istio/pkg/test/echo/common/response	[no test files]
?   	istio.io/istio/pkg/test/echo/common/scheme	[no test files]
?   	istio.io/istio/pkg/test/echo/proto	[no test files]
?   	istio.io/istio/pkg/test/echo/server	[no test files]
?   	istio.io/istio/pkg/test/echo/server/endpoint	[no test files]
?   	istio.io/istio/pkg/test/echo/server/forwarder	[no test files]
?   	istio.io/istio/pkg/test/env	[no test files]
?   	istio.io/istio/pkg/test/envoy	[no test files]
?   	istio.io/istio/pkg/test/fakes/gce_metadata_server	[no test files]
ok  	istio.io/istio/pkg/test/framework	0.365s
?   	istio.io/istio/pkg/test/framework/components/cluster	[no test files]
ok  	istio.io/istio/pkg/test/framework/components/cluster/clusterboot	(cached)
?   	istio.io/istio/pkg/test/framework/components/cluster/kube	[no test files]
ok  	istio.io/istio/pkg/test/framework/components/cluster/staticvm	(cached)
ok  	istio.io/istio/pkg/test/framework/components/echo	(cached)
ok  	istio.io/istio/pkg/test/framework/components/echo/cmd/echogen	(cached)
?   	istio.io/istio/pkg/test/framework/components/echo/common	[no test files]
?   	istio.io/istio/pkg/test/framework/components/echo/echoboot	[no test files]
ok  	istio.io/istio/pkg/test/framework/components/echo/echotest	(cached)
ok  	istio.io/istio/pkg/test/framework/components/echo/kube	(cached)
?   	istio.io/istio/pkg/test/framework/components/echo/staticvm	[no test files]
?   	istio.io/istio/pkg/test/framework/components/echo/util/traffic	[no test files]
?   	istio.io/istio/pkg/test/framework/components/environment/kube	[no test files]
?   	istio.io/istio/pkg/test/framework/components/gcemetadata	[no test files]
?   	istio.io/istio/pkg/test/framework/components/istio	[no test files]
?   	istio.io/istio/pkg/test/framework/components/istio/ingress	[no test files]
?   	istio.io/istio/pkg/test/framework/components/istioctl	[no test files]
ok  	istio.io/istio/pkg/test/framework/components/namespace	(cached)
?   	istio.io/istio/pkg/test/framework/components/opentelemetry	[no test files]
?   	istio.io/istio/pkg/test/framework/components/prometheus	[no test files]
?   	istio.io/istio/pkg/test/framework/components/redis	[no test files]
?   	istio.io/istio/pkg/test/framework/components/stackdriver	[no test files]
?   	istio.io/istio/pkg/test/framework/components/zipkin	[no test files]
?   	istio.io/istio/pkg/test/framework/config	[no test files]
?   	istio.io/istio/pkg/test/framework/errors	[no test files]
?   	istio.io/istio/pkg/test/framework/features	[no test files]
?   	istio.io/istio/pkg/test/framework/image	[no test files]
ok  	istio.io/istio/pkg/test/framework/integration	(cached)
ok  	istio.io/istio/pkg/test/framework/label	(cached)
ok  	istio.io/istio/pkg/test/framework/resource	(cached)
?   	istio.io/istio/pkg/test/framework/tools/featuresgen	[no test files]
ok  	istio.io/istio/pkg/test/framework/tools/featuresgen/cmd	(cached)
?   	istio.io/istio/pkg/test/helm	[no test files]
?   	istio.io/istio/pkg/test/kube	[no test files]
?   	istio.io/istio/pkg/test/scopes	[no test files]
?   	istio.io/istio/pkg/test/shell	[no test files]
?   	istio.io/istio/pkg/test/util/file	[no test files]
?   	istio.io/istio/pkg/test/util/reserveport	[no test files]
ok  	istio.io/istio/pkg/test/util/retry	(cached)
ok  	istio.io/istio/pkg/test/util/structpath	(cached)
?   	istio.io/istio/pkg/test/util/tmpl	[no test files]
ok  	istio.io/istio/pkg/test/util/yml	(cached)
?   	istio.io/istio/pkg/testcerts	[no test files]
ok  	istio.io/istio/pkg/uds	(cached)
ok  	istio.io/istio/pkg/url	(cached)
ok  	istio.io/istio/pkg/util/gogo	(cached)
?   	istio.io/istio/pkg/util/gogoprotomarshal	[no test files]
?   	istio.io/istio/pkg/util/istiomultierror	[no test files]
?   	istio.io/istio/pkg/util/protomarshal	[no test files]
ok  	istio.io/istio/pkg/util/shellescape	(cached)
ok  	istio.io/istio/pkg/util/strcase	(cached)
ok  	istio.io/istio/pkg/wasm	(cached)
ok  	istio.io/istio/pkg/webhooks	(cached)
ok  	istio.io/istio/pkg/webhooks/validation/controller	(cached)
ok  	istio.io/istio/pkg/webhooks/validation/server	(cached)
?   	istio.io/istio/samples/health-check	[no test files]
ok  	istio.io/istio/samples/jwt-server/src	(cached)
ok  	istio.io/istio/samples/tcp-echo/src	(cached)
?   	istio.io/istio/security/pkg/cmd	[no test files]
ok  	istio.io/istio/security/pkg/credentialfetcher	(cached)
ok  	istio.io/istio/security/pkg/credentialfetcher/plugin	(cached)
ok  	istio.io/istio/security/pkg/k8s	(cached)
ok  	istio.io/istio/security/pkg/k8s/chiron	(cached)
?   	istio.io/istio/security/pkg/k8s/controller	[no test files]
ok  	istio.io/istio/security/pkg/k8s/secret	(cached)
?   	istio.io/istio/security/pkg/k8s/tokenreview	[no test files]
?   	istio.io/istio/security/pkg/monitoring	[no test files]
ok  	istio.io/istio/security/pkg/nodeagent/cache	(cached)
ok  	istio.io/istio/security/pkg/nodeagent/caclient	(cached)
ok  	istio.io/istio/security/pkg/nodeagent/caclient/providers/citadel	(cached)
ok  	istio.io/istio/security/pkg/nodeagent/caclient/providers/google	(cached)
?   	istio.io/istio/security/pkg/nodeagent/caclient/providers/google/mock	[no test files]
?   	istio.io/istio/security/pkg/nodeagent/caclient/providers/mock	[no test files]
ok  	istio.io/istio/security/pkg/nodeagent/plugin/providers/google/stsclient	(cached)
ok  	istio.io/istio/security/pkg/nodeagent/sds	(cached)
?   	istio.io/istio/security/pkg/nodeagent/test	[no test files]
ok  	istio.io/istio/security/pkg/nodeagent/test/cert_rotation	(cached)
ok  	istio.io/istio/security/pkg/nodeagent/test/csr_failure	(cached)
ok  	istio.io/istio/security/pkg/nodeagent/test/empty_certchain	(cached)
?   	istio.io/istio/security/pkg/nodeagent/test/mock	[no test files]
ok  	istio.io/istio/security/pkg/nodeagent/test/success_sds	(cached)
?   	istio.io/istio/security/pkg/nodeagent/util	[no test files]
ok  	istio.io/istio/security/pkg/pki/ca	(cached)
?   	istio.io/istio/security/pkg/pki/ca/mock	[no test files]
ok  	istio.io/istio/security/pkg/pki/error	(cached)
ok  	istio.io/istio/security/pkg/pki/ra	(cached)
ok  	istio.io/istio/security/pkg/pki/util	(cached)
ok  	istio.io/istio/security/pkg/server/ca	(cached)
ok  	istio.io/istio/security/pkg/server/ca/authenticate	(cached)
ok  	istio.io/istio/security/pkg/server/ca/authenticate/kubeauth	(cached)
?   	istio.io/istio/security/pkg/stsservice	[no test files]
?   	istio.io/istio/security/pkg/stsservice/mock	[no test files]
ok  	istio.io/istio/security/pkg/stsservice/server	(cached)
?   	istio.io/istio/security/pkg/stsservice/test	[no test files]
ok  	istio.io/istio/security/pkg/stsservice/test/failure_sts_token_fetch	(cached)
ok  	istio.io/istio/security/pkg/stsservice/test/proxy_cached_sts_token	(cached)
ok  	istio.io/istio/security/pkg/stsservice/test/renew_sts_token	(cached)
ok  	istio.io/istio/security/pkg/stsservice/test/server_cached_short_lived_sts_token	5.102s
ok  	istio.io/istio/security/pkg/stsservice/test/server_cached_sts_token	4.866s
ok  	istio.io/istio/security/pkg/stsservice/test/sts_fetch_timeout	(cached)
ok  	istio.io/istio/security/pkg/stsservice/test/success_sts	(cached)
ok  	istio.io/istio/security/pkg/stsservice/tokenmanager	(cached)
ok  	istio.io/istio/security/pkg/stsservice/tokenmanager/google	(cached)
?   	istio.io/istio/security/pkg/stsservice/tokenmanager/google/mock	[no test files]
?   	istio.io/istio/security/pkg/testing/sdsc	[no test files]
ok  	istio.io/istio/security/pkg/util	(cached)
?   	istio.io/istio/security/pkg/util/mock	[no test files]
?   	istio.io/istio/security/proto/providers/google	[no test files]
?   	istio.io/istio/security/tools/generate_cert	[no test files]
?   	istio.io/istio/security/tools/generate_csr	[no test files]
?   	istio.io/istio/security/tools/sdsclient	[no test files]
ok  	istio.io/istio/tests/binary	(cached)
ok  	istio.io/istio/tests/common/jwt	(cached)
ok  	istio.io/istio/tests/fuzz	(cached)
?   	istio.io/istio/tests/util	[no test files]
ok  	istio.io/istio/tests/util/leak	(cached)
?   	istio.io/istio/tests/util/sanitycheck	[no test files]
?   	istio.io/istio/tools/bug-report	[no test files]
?   	istio.io/istio/tools/bug-report/pkg/archive	[no test files]
?   	istio.io/istio/tools/bug-report/pkg/bugreport	[no test files]
?   	istio.io/istio/tools/bug-report/pkg/cluster	[no test files]
?   	istio.io/istio/tools/bug-report/pkg/common	[no test files]
ok  	istio.io/istio/tools/bug-report/pkg/config	(cached)
?   	istio.io/istio/tools/bug-report/pkg/content	[no test files]
ok  	istio.io/istio/tools/bug-report/pkg/filter	(cached)
?   	istio.io/istio/tools/bug-report/pkg/kubeclient	[no test files]
?   	istio.io/istio/tools/bug-report/pkg/kubectlcmd	[no test files]
ok  	istio.io/istio/tools/bug-report/pkg/processlog	(cached)
?   	istio.io/istio/tools/bug-report/pkg/util/match	[no test files]
?   	istio.io/istio/tools/bug-report/pkg/util/path	[no test files]
?   	istio.io/istio/tools/istio-clean-iptables	[no test files]
?   	istio.io/istio/tools/istio-clean-iptables/pkg/cmd	[no test files]
?   	istio.io/istio/tools/istio-clean-iptables/pkg/config	[no test files]
?   	istio.io/istio/tools/istio-iptables	[no test files]
ok  	istio.io/istio/tools/istio-iptables/pkg/builder	(cached)
ok  	istio.io/istio/tools/istio-iptables/pkg/cmd	(cached)
?   	istio.io/istio/tools/istio-iptables/pkg/config	[no test files]
?   	istio.io/istio/tools/istio-iptables/pkg/constants	[no test files]
?   	istio.io/istio/tools/istio-iptables/pkg/dependencies	[no test files]
ok  	istio.io/istio/tools/istio-iptables/pkg/validation	(cached)
```
**Functionality validation:**

```
$ istioctl analyze -A
......
Info [IST0131] (VirtualService non-method-get.default) VirtualService rule "send product to sample.foo" match #1 of prefix /api/v1/products is not used (duplicate/overlapping match in rule #0 of prefix /api/v1/product on "send product to sample.foo").
Info [IST0131] (VirtualService overlapping-in-single-match.default) VirtualService rule "send product to sample.foo" match #1 of prefix /api/v1/products is not used (duplicate/overlapping match in rule #0 of prefix /api/v1/product on "send product to sample.foo").
Info [IST0131] (VirtualService overlapping-in-two-matches.default) VirtualService rule "send products to sample.bar" match #0 of prefix /api/v1/products is not used (duplicate/overlapping match in rule #0 of prefix /api/v1/product on "send product to sample.foo").
Info [IST0131] (VirtualService overlapping-mathes-with-different-methods.default) VirtualService rule "send products to sample.bar" match #0 of prefix /api/v1/product is not used (duplicate/overlapping match in rule #0 of prefix /api/v1/prod on "send product to sample.foo").
Error: Analyzers found issues when analyzing all namespaces.
See https://istio.io/v1.11/docs/reference/config/analysis for more information about causes and resolutions.
```

File: pkg/config/validation/validation.go [hunk 1]
```diff
@@ -752,11 +754,6 @@ var ValidateEnvoyFilter = registerValidateFunc("ValidateEnvoyFilter",
 					listenerMatch := cp.Match.GetListener()
 					if listenerMatch.FilterChain != nil {
 						if listenerMatch.FilterChain.Filter != nil {
-							if cp.ApplyTo == networking.EnvoyFilter_LISTENER || cp.ApplyTo == networking.EnvoyFilter_FILTER_CHAIN {
```
Comment (hzxuzhonghu) [inline pkg/config/validation/validation.go:755]: revert

Comment (zhlsunshine) [reply pkg/config/validation/validation.go]: okay, reverted

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**